src/com/palmergames/bukkit/towny/db/TownyDatabaseHandler.java

```diff
@@ -291,6 +290,27 @@ public TownyWorld getTownWorld(String townName) {
 	@Override
 	public void removeResident(Resident resident) {
 
+		// Remove resident from towns' outlawlists.
+		for (Town townOutlaw : getTowns()) {
+			if (townOutlaw.hasOutlaw(resident)) {
+				townOutlaw.removeOutlaw(resident);
+				saveTown(townOutlaw);
+			}
+		}
+
+		// Remove resident from residents' friendslists.
+		List<Resident> toSave = new ArrayList<>();
+		for (Resident toCheck : new ArrayList<>(universe.getResidentMap().values())) {
```

Comment:
Were you having `ConcurrentModicationExpection`'s here as well? If not this sould be just `universe.getResidentMap.values()`